Automattic/jetpack
# Social: Fix Reconnect link for broken connections in the editor from Calypso to connections modal

In the editor, we show a notice for broken connections and it [points to Calypso](https://github.com/Automattic/jetpack/blob/c2acdc2f71da5644ee08bf94c874f6336ed10f18/projects/js-packages/publicize-components/src/components/form/broken-connections-notice.tsx#L28)

## Proposed changes:

* For Jetpack sites, update the reconnect link from Calypso to connections management UI (modal)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:








<details>
    <summary>To simulate a broken connection, you can make these changes (click to expand)</summary>

```diff
diff --git a/projects/packages/publicize/src/class-publicize.php b/projects/packages/publicize/src/class-publicize.php
index d23c38d358..963d4a787d 100644
--- a/projects/packages/publicize/src/class-publicize.php
+++ b/projects/packages/publicize/src/class-publicize.php
@@ -287,7 +287,7 @@ class Publicize extends Publicize_Base {
                $connection_results_map = array();

                foreach ( $connection_results as $connection_result ) {
-                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'ok' : 'broken';
+                       $connection_results_map[ $connection_result['connection_id'] ] = $connection_result['test_success'] ? 'broken' : 'broken';
                }
                foreach ( $connections as $key => $connection ) {
                        if ( isset( $connection_results_map[ $connection['connection_id'] ] ) ) {
```
</details>


* Go to post editor
* Open Jetpack/Social sidebar
* Confirm that you see the broken connections notice
* Confirm that the message is correct and makes sense for both Jetpack sites and simple sites (for which the link will goot Calypso connections management)
* Click on the "connections management" link
* Confirm that it opens the connections modal instead of going to Calypso
* Test the change on Simple sites
* Confirm that the connections management link points to Calypso


<img width="299" alt="Screenshot 2024-06-14 at 11 33 24 AM" src="https://github.com/Automattic/jetpack/assets/18226415/688ca22b-f97d-4005-ae7a-ca06082c85e1">
